dartagnan/src/main/java/com/dat3m/dartagnan/parsers/program/visitors/VisitorLitmusAArch64.java

```diff
@@ -53,9 +57,32 @@ public Object visitMain(LitmusAArch64Parser.MainContext ctx) {
         visitVariableDeclaratorList(ctx.variableDeclaratorList());
         visitInstructionList(ctx.program().instructionList());
         VisitorLitmusAssertions.parseAssertions(programBuilder, ctx.assertionList(), ctx.assertionFilter());
-        return programBuilder.build();
+        Program prog = programBuilder.build();
+        replaceXZRRegister(prog);
+
+        return prog;
+
     }
 
+    /*
+    The "xzr" register plays a special role in AArhc64:
+      1. Reading accesses always return the value 0.
+      2. Discards data when written.
+     TODO: The below code is a simple fix to guarantee point 1. above.
+      Point 2. might also be resolved: although we do not prevent writing to xzr,
+      the value of xzr is never read after the transformation so its value is effectively 0.
+      However, the exists/forall clauses could still refer to that register and observe a non-zero value.
+     */
+    private void replaceXZRRegister(Program program) {
```

Comment:
Why not just also do the replacement on the specification?

Is that register only used in local computations or can it also be used in memory operations?
Specifically, if a load is performed into that register, then the load value is obviously discarded, but does the load otherwise still provide consistency guarantees (e.g. ACQ)? Will operations reading from the special register afterwards carry dependencies from the load?
If the latter is true, then replacing the register simply by `0` is unsound as it breaks data deps.

Reply:
> Why not just also do the replacement on the specification?

Done

> Is that register only used in local computations or can it also be used in memory operations? Specifically, if a load is performed into that register, then the load value is obviously discarded, but does the load otherwise still provide consistency guarantees (e.g. ACQ)? Will operations reading from the special register afterwards carry dependencies from the load? If the latter is true, then replacing the register simply by `0` is unsound as it breaks data deps.

I could not find any proper answer by taking a look to the documentation. The closer I got was
```
Register '31' is one of two registers depending on the instruction context:

For instructions dealing with the stack, it is the stack pointer, named rsp
For all other instructions, it is a "zero" register, which returns 0 when read and discards data when written - named rzr (xzr, wzr)
``` 
but it is not really clear which instructions could actually use the registers, i.e. the documentation of loads was not clear about xzr being allowed or not.

When I get some more time, I will try to come up with some litmus tests and check what herd7 says.

Reply:
I expected not much information to be out there, so that is fine :).
But I find the point about the stack pointer interesting. So in some contexts is actually represents a non-zero value.